src/main/groovy/com/google/protobuf/gradle/ToolsLocator.groovy

```diff
@@ -108,4 +108,13 @@ class ToolsLocator {
       }
     }
   }
+
+  static List<String> artifactParts(String artifactCoordinate) {
+    final validName = '[+-.0-9A-Z_a-z]+'
+
+    ((artifactCoordinate =~ /(${validName}):(${validName}):(${validName})(:${validName})?(@${validName})?/)[0]
```

Comment:
I do think that code is cleaner but judging from what I've been seeing from the codenarc output, it won't pass (and I haven't been able to reproduce locally the codenarc errors so it's a pain to deal with).